src/SearchWidget.cpp

```diff
@@ -0,0 +1,181 @@
+#include "SearchWidget.h"
+#include "ui_SearchForm.h"
+
+#include <iostream>
+
+#include "TextEdit.h"
+
+SearchWidget::SearchWidget(TextEdit* textEdit, QWidget *parent )
+    : QWidget( parent )
+    , mUi(new Ui::SearchForm)
+    , mTextEdit(textEdit)
+{
+    mUi->setupUi(this);
+    mUi->guiPreviousButton->setArrowType(Qt::UpArrow);
+    mUi->guiPreviousButton->setToolButtonStyle(Qt::ToolButtonTextBesideIcon);
+    mUi->guiNextButton->setArrowType(Qt::DownArrow);
+    mUi->guiNextButton->setToolButtonStyle(Qt::ToolButtonTextBesideIcon);
+
+    QObject::connect(mUi->guiNextButton, &QToolButton::clicked, this, &SearchWidget::onNextButtonClicked);
+    QObject::connect(mUi->guiPreviousButton, &QToolButton::clicked, this, &SearchWidget::onPreviousButtonClicked);
+    QObject::connect(mTextEdit, &TextEdit::textChanged, this, &SearchWidget::documentChange);
+    QObject::connect(mUi->guiSearchLine, &QLineEdit::textChanged, this, &SearchWidget::searchLineChanged);
+    QObject::connect(mUi->guiCloseButton, &QToolButton::clicked, this, &SearchWidget::closeSearch);
+}
+
+SearchWidget::~SearchWidget()
+{
+    delete mUi;
+}
+
+void SearchWidget::initialize(const QString & text)
+{
+    mSearchVisible = true;
+    mUi->guiSearchLine->setFocus();
+    mUi->guiSearchLine->setText(text);
+    searchWord(true);
+}
+
+void SearchWidget::uninitialize()
+{
+    mSearchVisible = false;
+    highLightedWords(false);
+}
+
+
+void SearchWidget::keyPressEvent(QKeyEvent * event)
+{
+    if (event->key() == Qt::Key_Return || event->key() == Qt::Key_Enter) {
+        onNextButtonClicked();
+    }
+}
+
+void SearchWidget::searchWord(bool selectNext, QString searchValue)
+{
+    if (searchValue.isEmpty()) {
+        searchValue = mUi->guiSearchLine->text();
+    }
+    mTextDocument = mTextEdit->toPlainText();
+    mCurrentSelectedWord = -1;
+    mPositionWords.clear();
```

Comment:
Clearing `mPositionWords` should be done by `searchPositionsWordsInDocument`, it's its job to update `mPositionWords`.

Reply:
resolve in commit 17c4d5b